Dockerfile

```diff
@@ -0,0 +1,25 @@
+# app/Dockerfile
+
+FROM python:3.9-slim
+
+WORKDIR /app
+
+RUN apt-get update && apt-get install -y \
+  build-essential \
+  ffmpeg \
+  libsm6 \
+  libxext6 \
+  curl \
+  software-properties-common \
+  git \
+  && rm -rf /var/lib/apt/lists/*
+
+RUN git clone https://github.com/wiktorlazarski/head-segmentation.git .
+
+RUN pip3 install -e .
+
+EXPOSE 8501
+
+HEALTHCHECK CMD curl --fail http://localhost:8501/_stcore/health
+
+ENTRYPOINT ["streamlit", "run", "apps/web_checking.py", "--server.port=8501", "--server.address=0.0.0.0"]
```

Comment:
Also a genuine curiosity question - previously you set a `WORKDIR` to apps. Doesn't it mean that you can skip `apps/` part in the entry point list 🤔? I would guess that something like would work
```suggestion
ENTRYPOINT ["streamlit", "run", "web_checking.py", "--server.port=8501", "--server.address=0.0.0.0"]
```
wdyt? Or maybe setting this `WORKDIR` isn't necessary 🤔?

Reply:
RIGHT! Usually formatters are taking care of that 😅

Reply:
`WORKDIR` creates a directory inside the container (and sets it as an operating one). When we perform `git clone` operation we are creating a structure in which `web_checking.py` is located inside `/app/apps/web_checking.py`. Since all of the command are executed from the context of `/app` (see [WORKDIR](https://docs.docker.com/engine/reference/builder/#workdir) reference) we have to execute `Entrypoint` command with `appps/web_checking.py` command

Reply:
Nice, thanks for doing that and detailed explanation 🙌.